romanesco/plugins/gaia/classes.py

```diff
@@ -0,0 +1,160 @@
+import json
+import sys
+import ast
+import os
+from romanesco.format import validators
+
+
+class RomanescoTypeConverter(object):
+    def __new__(cls, value, **kwargs):
+
+        if cls.is_valid(value, **kwargs):
+            return cls.convert(value, **kwargs)
+        else:
+            raise Exception("Invalid Romanesco Type")
+
+    @classmethod
+    def is_valid(cls, value, **kwargs):
+        # Currently we don't actually validate anything
+        return True
+
+    @classmethod
+    def convert(cls, value, **kwargs):
+        # Currently we don't actually convert anything
+        # Would have to resolve the format of value before
+        # converting it to the format in kwargs
+        return value
+
+# Define the Output converter
+Output = type("Output", (RomanescoTypeConverter,), {})
+
+romanesco_types = []
+
+module = sys.modules[__name__]
+
+for key, val in validators.items():
+    romanesco_types.append(key.title())
+
+    if not hasattr(module, key.title()):
+        setattr(module,
+                str(key.title()),
+                type(str(key.title()), (RomanescoTypeConverter, ), {}))
```

Comment:
Could we make `RomanescoTypeConverter` a metaclass that registers its subclasses in `romanesco_types` for us? Also couldn't `romanesco_types` be made into a dict rather than a list so we could just store all the dynamic classes within the dict rather than within `module.__dict__`?

(Also the loop header could just be rewritten `for key in validators.keys():`)